### PR TITLE
chore: upgrade go to 1.22, rebuild base image

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
upgrade go to 1.22, rebuild base image

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  upgrade go to 1.22, rebuild base image            |
| 🇨🇳 中文    |     升级go版本到1.22，重新构建base镜像         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: build/dockerfiles/base/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FROM bitnami/git:latest AS git-from
 
-FROM --platform=$TARGETPLATFORM golang:1.19-bookworm
+FROM --platform=$TARGETPLATFORM golang:1.22.4-bookworm
```
Comment (iutx) [inline build/dockerfiles/base/Dockerfile:3]: Is using `1.22` better?

**Review by iutx — DISMISSED** [review]
@chengjoey

File: build/dockerfiles/base/Dockerfile (hunk 1 above)
Comment (chengjoey) [reply build/dockerfiles/base/Dockerfile]: right, [1.22](https://github.com/docker-library/golang/blob/58bc678a838d467ca5afb7e0ee636cf5b8459da2/1.22/bookworm/Dockerfile) currently using 1.22.4, using `1.22` can automatically upgrade to a higher version without changing `Dockerfile`, done~

File: pkg/jsonstore/etcd/etcd.go [hunk 2]
```diff
@@ -170,22 +170,24 @@ func (s *Store) retry(do func(cli *clientv3.Client) (interface{}, error)) (inter
 		respC1 <- resp
 	}()
 
+	var secondClient *clientv3.Client
```
Comment (sfwn) [inline pkg/jsonstore/etcd/etcd.go:173]: 6

File: go.sum [hunk 3]
```diff
@@ -34,67 +32,752 @@ cloud.google.com/go v0.93.3/go.mod h1:8utlLll2EF5XMAV15woO4lSbWQlk8rer9aLOfLh7+Y
 cloud.google.com/go v0.94.1/go.mod h1:qAlAugsXlC+JWO+Bke5vCtc9ONxjQT3drlTTnAplMW4=
 cloud.google.com/go v0.97.0/go.mod h1:GF7l59pYBVlXQIBLx3a761cZ41F9bBH3JUlihCt2Udc=
 cloud.google.com/go v0.99.0/go.mod h1:w0Xx2nLzqWJPuozYQX+hFfCSI8WioryfRDzkoI/Y2ZA=
+cloud.google.com/go v0.100.1/go.mod h1:fs4QogzfH5n2pBXBP9vRiU+eCny7lD2vmFZy79Iuw1U=
 cloud.google.com/go v0.100.2/go.mod h1:4Xra9TjzAeYHrl5+oeLlzbM2k3mjVhZh4UqTZ//w99A=
 cloud.google.com/go v0.102.0/go.mod h1:oWcCzKlqJ5zgHQt9YsaeTY9KzIvjyy0ArmiBUgpQ+nc=
-cloud.google.com/go v0.102.1 h1:vpK6iQWv/2uUeFJth4/cBHsQAGjn1iIE6AAlxipRaA0=
 cloud.google.com/go v0.102.1/go.mod h1:XZ77E9qnTEnrgEOvr4xzfdX5TRo7fB4T2F4O6+34hIU=
+cloud.google.com/go v0.104.0/go.mod h1:OO6xxXdJyvuJPcEPBLN9BJPD+jep5G1+2U5B5gkRYtA=
```
Comment (sfwn) [inline go.sum:39]: polish it

**Review by sfwn — CHANGES_REQUESTED** [review]

**Pull request merged**